apis/run/v1alpha3/kappcontrollerconfig_types.go

```diff
@@ -13,24 +13,28 @@ import (
 type KappControllerConfigSpec struct {
 	// The namespace in which kapp-controller is deployed
 	//+kubebuilder:validation:Optional
-	//+kubebuilder:default:=kube-system
+	//+kubebuilder:default:=tkg-system
 	Namespace string `json:"namespace,omitempty"`
 
-	KappController KappController `json:"kappController,omitempty"`
+	//+kubebuilder:validation:Optional
+	//+kubebuilder:default:={deployment:{hostNetwork:true}}
+	KappController KappController `json:"kappController,omitempty"` // object default needs at least one param so that CRD generation is not null, issue https://github.com/kubernetes-sigs/controller-tools/issues/631
 }
 
 type KappController struct {
 	// Whether to create namespace specified for kapp-controller
 	//+kubebuilder:validation:Optional
-	//+kubebuilder:default:=true
+	//+kubebuilder:default:=false
 	CreateNamespace bool `json:"createNamespace,omitempty"`
```

Comment:
**Question**: Does the code create this namespace ?

Reply:
Just [looked up](https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.34.0/bundle/config/overlays/change-namespace.yaml#L12), it removes the default kapp-controller namespace if false.

I noticed there is also https://github.com/vmware-tanzu/community-edition/blob/main/addons/packages/kapp-controller/0.34.0/bundle/config/overlays/change-namespace.yaml#L58 which will end up creating namespace if globalNamespace and namespace are same.

Reply:
Logged bug https://github.com/vmware-tanzu/tanzu-framework/issues/1944